dev-embedded/sdcc/sdcc-4.2.0.ebuild

```diff
@@ -0,0 +1,167 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit autotools
+
+if [[ ${PV} == "9999" ]] ; then
+	ESVN_REPO_URI="https://svn.code.sf.net/p/sdcc/code/trunk/sdcc"
+	inherit subversion
+else
+	SRC_URI="
+		mirror://sourceforge/sdcc/${PN}-src-${PV}.tar.bz2
+		doc? ( mirror://sourceforge/sdcc/${PN}-doc-${PV}.tar.bz2 )
+	"
+
+	KEYWORDS="~amd64 ~x86"
+fi
+
+DESCRIPTION="Small device C compiler (for various microprocessors)"
+HOMEPAGE="http://sdcc.sourceforge.net/"
+
+LICENSE="
+	GPL-2 ZLIB
+	non-free? ( MicroChip-SDCC )
+	packihx? ( public-domain )
+"
+SLOT="0"
+# in order of configure.ac's AC_DO_PORT stanzas
+SDCC_PORTS="
+	mcs51
+	z80 z180
+	r2k r2ka r3ka
+	sm83
+	tlcs90
+	ez80-z80
+	z80n
+	ds390 ds400
+	pic14 pic16
+	hc08
+	s08
+	stm8
+	pdk13 pdk14 pdk15 pdk16
+	mos6502 mos65c02
+"
+IUSE="
+	${SDCC_PORTS}
+	+boehm-gc device-lib doc non-free packihx +sdbinutils sdcdb +sdcpp ucsim
+"
+
+for port in ${SDCC_PORTS}; do
+REQUIRED_USE="${REQUIRED_USE}
+	${port}? ( sdbinutils )
+"
+done
+REQUIRED_USE="${REQUIRED_USE}
+	|| ( ${SDCC_PORTS} )
+"
+
+RDEPEND="
+	dev-libs/boost:=
+	pic14? ( >=dev-embedded/gputils-0.13.7 )
+	pic16? ( >=dev-embedded/gputils-0.13.7 )
+	boehm-gc? ( dev-libs/boehm-gc:= )
+	sdbinutils? ( sys-libs/zlib:= )
+	sdcdb? ( sys-libs/readline:0= )
+	ucsim? ( sys-libs/ncurses:= )
+"
+DEPEND="
+	${RDEPEND}
+	dev-util/gperf
+"
+PATCHES=(
+	"${FILESDIR}"/sdcc-3.8.0-override-override.patch
```

Comment:
This does not apply cleanly in the live ebuild, so it cannot be built with automake 2.71.

Will not fix it now, but let's keep it in mind.